lib/msf/core/post/linux/system.rb

```diff
@@ -102,6 +102,166 @@ def get_sysinfo
+    rescue
+      raise 'Could not gather listening ports'
+    end
+  end
+
+  #
+  # Gathers all SUID files on the filesystem.
+  # NOTE: This uses the Linux `find` command. It will most likely take a while to get all files.
+  # Consider specifying a more narrow find path.
+  # @param findpath The path on the system to start searching
+  # @return [Array]
+  def get_suid_files(findpath = '/')
+    cmd_exec("find #{findpath} -perm -4000 -print -xdev").to_s.split("\n")
+  rescue
+    raise "Could not retrieve all SUID files"
+  end
+
+  #
+  # Gets the $PATH environment variable
+  #
+  def get_path
+    cmd_exec('echo $PATH').to_s
+  rescue
+    raise "Unable to determine path"
+  end
+
+  #
+  # Gets basic information about the system's CPU.
+  # @return [Hash]
+  #
+  def get_cpu_info
+    info = {}
+    orig = cmd_exec("cat /proc/cpuinfo").to_s
+    cpuinfo = orig.split("\n\n")[0]
+    # This is probably a more platform independent way to parse the results (compared to splitting and assigning preset indices to values)
+    cpuinfo.split("\n").each do |l|
+      info[:speed_mhz]   = l.split(': ')[1].to_i if l =~ /cpu MHz/
+      info[:product]     = l.split(': ')[1]      if l =~ /model name/
+      info[:vendor]      = l.split(': ')[1]      if l =~ /vendor_id/
+    end
+    info[:cores] = orig.split("\n\n").size
+    info
+  rescue
+    raise "Could not get CPU information"
+  end
+
+  #
+  # Gets the hostname of the system
+  # @return [String]
+  #
+  def get_hostname
+    cmd_exec('uname -n').to_s
+  rescue
+    raise 'Unable to retrieve hostname'
+  end
+
+  #
+  # Gets the name of the current shell
+  # @return [String]
+  #
+  def get_shell_name
+    psout = cmd_exec('ps -p $$').to_s
+    psout.split("\n").last.split(' ')[3]
+  rescue
+    raise 'Unable to gather shell name'
+  end
+
+  #
+  # Checks if the system has gcc installed
+  # @return [Boolean]
+  #
+  def has_gcc?
+    command_exists? 'gcc'
+  rescue
+    raise 'Unable to check for gcc'
+  end
+
+  #
+  # Gets the version of gcc
+  # @return [String]
+  #
+  def gcc_version
+    return cmd_exec('gcc --version | head -n 1').to_s.split(' ')[3] if has_gcc?
+    raise 'System does not have gcc installed'
+  rescue
+    raise 'Unable to get gcc version'
+  end
+
+  #
+  # Checks if the `cmd` is installed on the system
+  # @return [Boolean]
+  #
+  def command_exists?(cmd)
+    cmd_exec("command -v #{cmd} && echo true").to_s.include? 'true'
+  rescue
+    raise "Unable to check if command `#{cmd}` exists"
+  end
+
+  #
+  # Gets the process id(s) of `program`
+  # @return [Array]
+  #
+  def pidof(program)
+    pids = []
+    full = cmd_exec('ps aux').to_s
+    full.split("\n").each do |pid|
+      pids << pid.split(' ')[1].to_i if pid =~ /#{program}/
+    end
+    pids
+  end
+
+  #
+  # Checks if `mount_path` is mounted with noexec
+  # @return [Boolean]
+  #
+  def noexec?(mount_path)
+    cmd_exec("mount | grep --color=never ' #{mount_path} '").to_s.include? 'noexec'
```

Comment:
Same reason as before. Why not `/proc/mounts`? :)

https://www.kernel.org/doc/Documentation/filesystems/proc.txt

Reply:
just a weird aside, `mount` appears to read /proc/self/mountinfo instead

Reply:
I think it's worth checking `mountinfo` if it's present!

Reply:
```
wvu@hiigara:~$ ll /proc/mounts
lrwxrwxrwx 1 root root 11 Apr 10 21:15 /proc/mounts -> self/mounts
wvu@hiigara:~$
```

Or we can rely on backward compatibility. :P

Reply:
```
wvu@hiigara:~$ ll /proc/mounts
lrwxrwxrwx 1 root root 11 Apr 10 21:25 /proc/mounts -> self/mounts
wvu@hiigara:~$
```

Or we can rely on backward compatibility. :P

Reply:
But I think we need just `noexec` or `nosuid`, so it's kinda a moot point for these two methods.

Reply:
So really this depends on how much we care about mount namespaces and whether to independently check for `/proc/self/mountinfo` or `/proc/self/mounts` or rely on `/proc/mounts` to be a symlink to the latter. FWIW, `/etc/mtab` is also a symlink to `/proc/self/mounts`.

Reply:
I feel like enumerating the edge cases while writing these libraries is something of a rabbit hole. The code works for the major cases. :)

Reply:
I've already made the change for parsing `/proc/mounts`, I'm testing it now. If it should default to `/proc/self/mounts` it's a pretty easy change.